src/Cli/dotnet/Commands/Run/RunCommand.cs

```diff
@@ -355,7 +355,7 @@ static void ValidatePreconditions(ProjectInstance project)
 
         static RunProperties ReadRunPropertiesFromProject(ProjectInstance project, string[] applicationArgs)
         {
-            var runProperties = RunProperties.FromProjectAndApplicationArguments(project, applicationArgs, fallbackToTargetPath: false);
```

Comment:
`dotnet run` never falls back to `TargetPath`, and I don't see any valid scenario for falling back to `TargetPath`. So not doing it anymore.